active-rfcs/0000-vtu-deprecations.md

```diff
@@ -0,0 +1,188 @@
+# API Deprecation [RFC]
+
+Created: Mar 28, 2020 7:39 PM
+
+- Start Date: 2020.03.28
+- Target Major Version: 1.x
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+Deprecate lesser used or bad practice methods and properties from the upcoming VueTestUtils 1.x API, aimed at Vue 3. 
+
+## Motivation
+
+With the current version of Vue Test Utils-beta.x, we notice the API is large and somewhat confusing to most  users, which are not well acquainted with testing Vue apps, leading to bad testing practices. 
+
+This PR aims to trim down the VTU API, replacing listed methods with improved docs, more examples for all the related cases or improved testing methods.
+
+## Detailed design
+
+### Methods
+
+**Wrapper.emittedByOrder -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#emittedbyorder)
+
+Rarely used, use `emitted` instead.
+
+```js
+expect(wrapper.emitted('change')[0]).toEqual(['param1', 'param2'])
+```
+
+**get -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#get) 
+
+Recently added, throws error if nothing is matched. Will be merged into `find`.
+
+```js
+expect(wrapper.find('.notExisting')).toThrow()
+
+const element = wrapper.find('.notExisting') // will throw error
+```
+
+**is -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#is)
+
+Not that much useful. Use `element.tagName` instead. 
+
+```js
+expect(wrapper.element.tagName).toEqual('div')
+```
+
+**isEmpty -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isempty)
+
+Use custom matcher like [jest-dom](https://github.com/testing-library/jest-dom#tobeempty) on the element
+
+```js
+expect(wrapper.element).toBeEmpty()
+```
+
+**isVisible** - [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isvisible)
+
+Use custom matcher like [jest-dom](https://github.com/testing-library/jest-dom#tobevisible)
+
+```js
+expect(wrapper.element).toBeVisible()
+```
+
+**isVueInstance -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isvueinstance)
+
+Rarely used, no benefits.
+
+**props -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#props)
```

Comment:
Sometimes we want to test a component in isolation without testing its children. In this case the component’s boundary ends at the props it is passing its children, so shallow mounting and then calling this method to check the child component props is very useful.